graspy/inference/latent_distribution_test.py

```diff
@@ -34,95 +37,92 @@ class LatentDistributionTest(BaseInference):
 
-    def __init__(self, n_components=None, n_bootstraps=200, bandwidth=None):
+    # TODO: reference Varjavand paper when it is on arxiv
+
+    def __init__(
+        self, n_components=None, n_bootstraps=200, method="mgc", pass_graph=True
+    ):
         if n_components is not None:
             if not isinstance(n_components, int):
                 msg = "n_components must an int, not {}.".format(type(n_components))
                 raise TypeError(msg)
-
-        if not isinstance(n_bootstraps, int):
-            msg = "n_bootstraps must an int, not {}".format(type(n_bootstraps))
+        if type(n_bootstraps) is not int:
+            msg = "n_bootstraps must be an int, not {}".format(type(method))
             raise TypeError(msg)
-        elif n_bootstraps < 1:
-            msg = "{} is invalid number of bootstraps, must be greater than 1"
-            raise ValueError(msg.format(n_bootstraps))
-
-        if bandwidth is not None and not isinstance(bandwidth, float):
-            msg = "bandwidth must an int, not {}".format(type(bandwidth))
+        if n_bootstraps <= 0:
+            msg = "n_bootstraps must be > 0, not {}".format(n_bootstraps)
+            raise ValueError(msg)
+        if type(method) is not str:
+            msg = "method must be a string, not {}.".format(type(method))
             raise TypeError(msg)
-
-        super().__init__(embedding="ase", n_components=n_components)
+        if method not in ["mgc", "dcorr"]:
+            msg = "{} is not a valid test, must be mgc or dcorr.".format(method)
+            raise ValueError(msg)
+        super().__init__(
+            embedding="ase", n_components=n_components, pass_graph=pass_graph
+        )
+        self.method = method
+        self.symmetry = None
         self.n_bootstraps = n_bootstraps
-        self.bandwidth = bandwidth
-
-    def _gaussian_covariance(self, X, Y):
-        diffs = np.expand_dims(X, 1) - np.expand_dims(Y, 0)
-        if self.bandwidth is None:
-            self.bandwidth = 0.5
-        return np.exp(-0.5 * np.sum(diffs ** 2, axis=2) / self.bandwidth ** 2)
-
-    def _statistic(self, X, Y):
-        N, _ = X.shape
-        M, _ = Y.shape
-        x_stat = np.sum(self._gaussian_covariance(X, X) - np.eye(N)) / (N * (N - 1))
-        y_stat = np.sum(self._gaussian_covariance(Y, Y) - np.eye(M)) / (M * (M - 1))
-        xy_stat = np.sum(self._gaussian_covariance(X, Y)) / (N * M)
-        return x_stat - 2 * xy_stat + y_stat
+
+    def _k_sample_transform(self, x, y):
+        if x.shape[0] != y.shape[0]:
+            logging.warning(
+                "Results are not to be trusted for small N1/N2, or N1 not approximately N2!"
+            )
+        u = np.concatenate([x, y], axis=0)
+        v = np.concatenate([np.repeat(1, x.shape[0]), np.repeat(2, y.shape[0])], axis=0)
+        if u.ndim == 1:
+            u = u[..., np.newaxis]
+        if v.ndim == 1:
+            v = v[..., np.newaxis]
+        return u, v
 
     def _embed(self, A1, A2):
```

Comment:
is there anything fundamentally different about the embed method for this and the embed method for lpt, other than that one can operate on different n_verts?

Reply:
not really. I think ldt handles asymmetric as well?

Reply:
I don't believe it does at the moment, you could check. We could just as well concatenate left and right in the asymmetric case for LPT, we just haven't tested that a ton but should work as well as it does for LDT. If you'd like to fix that in the scope of this PR, go for it and refactor the embed method. If you don't want that to be in the scope of this PR, that's fine, but could you just make an issue saying "refactor embed method for LPT and LDT" or something like that?